server/lib/config/pbench-server-default.cfg

```diff
@@ -181,6 +181,10 @@ tasks = pbench-backup-tarballs
 # This is a template that is expanded by pbench-server-activate-create-crontab
 crontab =  * * * * *  flock -n %(lock-dir)s/pbench-prep-$SHIM_VERSION.lock %(script-dir)s/pbench-server-prep-shim-$SHIM_VERSION
 
+[pbench-put-shim]
```

Comment:
I suspect the reason Ansible isn't configuring this in your `crontab` is that it's not listed in `tasks` under the `pbench-results` role.

And for @ndokos to think about:

This raises an interesting question: do we want to handle this new server configuration by adding a *role*, perhaps something like `pbench-repeater`, which might have a `tasks` list with just `pbench-server-put-shim` and `pbench-dispatch`? Then the "normal" server configuration just wouldn't have a put-shim task, while the new "end of life" 0.69 default would have only put-shim ... without breaking a `pbench-results` (or any other role) deployment should someone really want to do it?

Reply:
yes, it needed to be placed in the task. Thank you, Dave:)

What if we put put-shim in `tasks` of `pbench-prep` rather than `tasks` of `pbench-results`?

Reply:
I'd like to hear @ndokos or @portante interpret exactly what the intended purpose was for the `pbench-prep` role and `pbench-prep-task`. This looks weird to me, as `pbench-prep-task` seems to be the mechanism for getting the prep-shim into crontab; but that task isn't listed under the "normal" `pbench-results` role. I don't quite see how that gets into `crontab` normally, but maybe it'd be clear if I spent the time analyzing the 0.69 utility that constructs the `crontab` from this info. (I've only looked at the `main` version, and not thoroughly.)

I'd also like some feedback on the "interesting question" I raised earlier: do we *break* the end-of-line 0.69 to work *only* as a repeater for 0.72, or do we use roles and tasks to simply create a new mode of operation that would also allow "normal" use? (We don't care, I think, and I don't know that anyone else is maintaining a 0.69 server or would have any interest in continuing to do so ... and we certainly don't want to encourage that.)

Reply:
My vote would be to tie off 0.69 and make it dependent upon having a separate 0.72 server.  I think we ultimately want to EOL 0.69, and anyone who wants to run it "on their own" can always branch from 0.69.9 instead of 0.69.10.

Reply:
Better late than never? The pbench-prep role is separate from the pbench-results role, because satellites run *only* pbench-prep, whereas the main server is running both pbench-prep and pbench-results.

The interesting question:
```
do we want to handle this new server configuration by adding a role, perhaps something like pbench-repeater, which might have a tasks list with just pbench-server-put-shim and pbench-dispatch? Then the "normal" server configuration just wouldn't have a put-shim task, while the new "end of life" 0.69 default would have only put-shim ... without breaking a pbench-results (or any other role) deployment should someone really want to do it?
```
I think it's just the put-shim that needs to run: no pbench-dispatch.  If the node that's running the put-shim is a completely separate node, it would have a put-shim task and no others.

Reply:
I would support having the put-shim run by a completely parallel set of structures, so that, if/when we strip away everything that isn't the put-shim, the break is clean and not entangled with anything which is supposed to go away.  As Nick said, the put-shim is supposed to run _instead_ of the rest of the server, so having it deployed by roles/tasks which are separate from the ones used for the server seems like a good thing.

However, it would be good if we gave at least a small consideration to how/where the put-shim will be deployed.  Can/would it be deployed on the same node as the server?  If so, we don't want to set ourselves up for a conflict, there.  Nevertheless, I don't think that any such conflict will result from the way that this file is structured...right?